components/integrations_marketplace/marketplace_modal.jsx

```diff
@@ -0,0 +1,204 @@
+        if (pluginsArray.length === 0) {
+            let noPluginsMessage = (
+                <FormattedMessage
+                    id='marketplace_modal.no_plugins'
+                    defaultMessage='There are no plugins available at this time.'
+                />);
+            if (installedList) {
+                noPluginsMessage = (
+                    <FormattedMessage
+                        id='marketplace_modal.no_plugins_installed'
+                        defaultMessage='You do not have any plugins installed.'
+                    />);
+            }
+
+            return (<div className='no_plugins_div'>
+                <br/>
+                <PluginIcon className='icon__plugin'/>
+                <br/>
+                <br/>
+                {noPluginsMessage}
+                <br/>
+                <br/>
+                {installedList ? (
+                    <a onClick={() => this.changeTab(MarketplaceTabs.ALL_PLUGINS)}>
+                        <FormattedMessage
+                            id='marketplace_modal.install_plugins'
+                            defaultMessage='Install Plugins'
+                        />
+                    </a>
+                ) : null
+                }
+            </div>);
+        }
+
+        return (<div className='more-modal__list'>
+            {
+                pluginsArray.map((p) => {
+                    return (
+                        <MarketplaceItem
+                            key={p.Manifest.id}
+                            id={p.Manifest.id}
+                            name={p.Manifest.name}
+                            description={p.Manifest.description}
+                            version={p.Manifest.version}
+                            isPrepackaged={false}
+                            downloadUrl={p.DownloadURL}
+                            homepageUrl={p.HomepageURL}
+                            installed={p.InstalledVersion !== ''}
+                            onConfigure={this.close}
+                            onInstalled={this.getMarketplacePlugins}
+                        />);
+                })
+            }
+        </div>);
+    }
+
+    render() {
+        //WIP: To add pagination section
+        const input = (
+            <div className='filter-row filter-row--full'>
+                <div className='col-sm-12'>
+                    <QuickInput
+                        id='searchChannelsTextbox'
+                        ref='filter'
+                        className='form-control filter-textbox search_input'
+                        placeholder={{id: t('marketplace_modal.search'), defaultMessage: 'Search Plugins'}}
+                        inputComponent={LocalizedInput}
+                        onInput={this.doSearch}
+                    />
+                </div>
+            </div>
+        );
+
+        let errorBanner = null;
+        if (this.props.serverError) {
+            errorBanner = (
```

Comment:
really surprising that we don't have an error banner, thought we did though, maybe with a different styling?

Reply:
There is an announcement banner but styling is different, specially because it's styled to be used in the top of the whole app. For marketplace, it was needed in the full-screen modal so this `div` was a simple solution for our needs.